index.js

```diff
@@ -12,8 +17,15 @@ function init() {
 
 	const selectedText = editor.getSelectedText();
 	const text = selectedText || editor.getText();
+	const config = atom.config.get('perfectionist');
 
-	postcss(perfectionist(atom.config.get('perfectionist'))).process(text, {
+	if (editor.getGrammar().scopeName === 'source.css.scss') {
```

Comment:
Okay, I made this change, but I don't understand what you mean by working with unsaved buffers. If it is unsaved, I assume the type is not known is what you are saying? If that is true, we're now running the scss syntax when there is an unsaved css file, which would be incorrect, right?